docs/classes.md

```diff
@@ -0,0 +1,263 @@
+-   The first parameter of the `Destructor Method` should always be named `self`.
+
+### - Python Example
+
+```python
+import numpy as np
+class MyClass:
+    def __init__(self : 'MyClass', param1 : 'int', param2 : 'float[:]'):
+        self.param1 = param1
+        self.param2 = param2
+        print("Object created!")
+        print(param1, param2)
+
+if __name__ == "__main__":
+    obj = MyClass(1, np.ones(4))
+```
+
+### PYTHON _OUTPUT_
+```Shell
+Object created!
+1 [1. 1. 1. 1.]
+```
+
+### - Header File Equivalent
+
+```C
+#ifndef SBOOF_H
+#define SBOOF_H
+
+struct MyClass {
+    int64_t param1;
+    t_ndarray param2;
+    bool is_freed;
+};
+
+void MyClass__init__(struct MyClass* self, int64_t param1, t_ndarray param2);
+void MyClass__del__(struct MyClass* self);
+
+#endif // sboof_H
+```
+### - C File Equivalent
+
+```C
+/*........................................*/
+void MyClass__init__(struct MyClass* self, int64_t param1, t_ndarray param2)
+{
+    self->is_freed = 0;
+    self->param1 = param1;
+    alias_assign(&self->param2, param2);
+    printf("%s\n", "Object created!");
+}
+/*........................................*/
+/*........................................*/
+void MyClass__del__(struct MyClass* self)
+{
+    if (!self->is_freed)
+    {
+        // pass
+        free_pointer(&self->param2);
+        self->is_freed = 1;
+    }
+}
+/*........................................*/
+```
+
+### - C Program File Equivalent
+```C
+int main()
+{
+    t_ndarray Dummy_0000 = {.shape = NULL};
+    struct MyClass obj;
+    Dummy_0000 = array_create(1, (int64_t[]){INT64_C(1)}, nd_double, false, order_c);
+    array_fill((double)1.0, Dummy_0000);
+    MyClass__init__(&obj, INT64_C(1), Dummy_0000);
+    free_array(&Dummy_0000);
+    MyClass__del__(&obj);
+    return 0;
+}
+```
+
+### C/VALGRIND _OUTPUT_
+
+```Shell
+Object created!
+1 [1.000000000000]
+
+==151606== Memcheck, a memory error detector
+==151606== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
+==151606== Using Valgrind-3.18.1 and LibVEX; rerun with -h for copyright info
+==151606== Command: ./MyClass
+==151606==
+Object created!
+1 [1.000000000000]
+==151606==
+==151606== HEAP SUMMARY:
+==151606==     in use at exit: 0 bytes in 0 blocks
+==151606==   total heap usage: 6 allocs, 6 frees, 1,064 bytes allocated
+==151606==
+==151606== All heap blocks were freed -- no leaks are possible
+==151606==
+==151606== For lists of detected and suppressed errors, rerun with: -s
+==151606== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0
+```
+
+## Class Methods
+
+-   Pyccel now supports Class Methods and Interfaces.
+-   In Pyccel, class attributes can be initialized within any method of the class.
+-   Pyccel enables classes to be passed as arguments to methods and functions and returned with modified data.
+-   The first parameter of the Class Method should always be named `self`.
```

Comment:
I think it may be worth moving this further up the file and grouping the examples. As they are quite detailed it could be easy to miss this section in the middle

Reply:
Fixed https://github.com/pyccel/pyccel/pull/1580/commits/65baee9d82decc0a4f2026777e35685a2ded628a.
Fixed https://github.com/pyccel/pyccel/pull/1580/commits/2cbad56fd8f824b2e827ac6512a991e661baa2a9.